plugins/CoreAdminHome/API.php

```diff
@@ -240,6 +243,41 @@ public function getTrackingFailures()
         return $failures;
     }
 
+    /**
+     * @param $idSite
+     * @param $period
+     * @param $date
+     * @param bool $segment
+     * @param bool $plugin
+     * @param bool $report
+     * @return mixed
+     * @throws \Piwik\Exception\UnexpectedWebsiteFoundException
+     * @internal
+     */
+    public function archiveReports($idSite, $period, $date, $segment = false, $plugin = false, $report = false)
+    {
+        // if cron archiving is running, we will invalidate in CronArchive, not here
+        $isArchivePhpTriggered = SettingsServer::isArchivePhpTriggered();
```

Comment:
do we need to check for super user permissions here?

Reply:
todo: need to check later how we can prevent a superuser calling this method on the cloud directly. But it should still be callable for the archiving. to be seen... might be only a matter of checking again that it won't trigger an archive if it was archived recently within TTL and that it won't simply invalidate the report

Reply:
I see core is using this method now as well and I suppose when browser archiving is enabled then a regular user might need to be allowed to call this. 

I wonder if we'd need to do something like `if (\Piwik\API\Request::getRootApiRequestMethod() === 'CoreAdminHome.archiveReports') { Piwik::checkUserHasSuperUserAccess() }`?

Might be generally also good to check always at least `Piwik::checkUserHasViewAccess($idSite);`? So a user can't trigger this method for like any date etc.

Reply:
`Piwik::checkUserHasViewAccess($idSite);` should definitely be called. I don't think we can  do `Piwik::checkUserHasSuperUserAccess()` since then Archive.php would need to set superuser access. I guess we can use the `getRootApiRequestMethod()` check... can't think of an issue w/ that.